src/vm/runners/builtin_runner/signature.rs

```diff
@@ -0,0 +1,469 @@
+            vec![MaybeRelocatable::from((self.base, 0))]
+        } else {
+            vec![]
+        }
+    }
+
+    pub fn base(&self) -> isize {
+        self.base
+    }
+    pub fn add_validation_rule(&self, memory: &mut Memory) -> Result<(), RunnerError> {
+        let cells_per_instance = self.cells_per_instance;
+        let signatures = self.signatures.clone();
+        let rule: ValidationRule = ValidationRule(Box::new(
+            move |memory: &Memory,
+                  address: &MaybeRelocatable|
+                  -> Result<Vec<MaybeRelocatable>, MemoryError> {
+                let address = match address {
+                    MaybeRelocatable::RelocatableValue(address) => address,
+                    _ => return Err(MemoryError::AddressNotRelocatable),
+                };
+
+                let address_offset = address.offset.mod_floor(&(cells_per_instance as usize));
+                let mem_addr_sum = memory.get(&(address + 1_i32));
+                let mem_addr_less = if address.offset > 0 {
+                    memory.get(
+                        &address
+                            .sub(1)
+                            .map_err(|_| MemoryError::AddressNotRelocatable)?,
+                    )
+                } else {
+                    Ok(None)
+                };
+                let (pubkey_addr, msg_addr) = match (address_offset, mem_addr_sum, mem_addr_less) {
+                    (0, Ok(Some(_element)), _) => {
+                        let pubkey_addr = address.clone();
+                        let msg_addr = address + 1_i32;
+                        (pubkey_addr, msg_addr)
+                    }
+                    (1, _, Ok(Some(_element))) if address.offset > 0 => {
+                        let pubkey_addr = address
+                            .sub(1)
+                            .map_err(|_| MemoryError::AddressNotRelocatable)?;
+                        let msg_addr = address.clone();
+                        (pubkey_addr, msg_addr)
+                    }
+                    _ => return Ok(Vec::new()),
+                };
+
+                let (_sign, msg) = memory
+                    .get_integer(&msg_addr)
+                    .map_err(|_| MemoryError::AddressNotRelocatable)?
+                    .to_bytes_be();
+                let (_sign, pubkey) = memory
+                    .get_integer(&pubkey_addr)
+                    .map_err(|_| MemoryError::AddressNotRelocatable)?
+                    .to_bytes_be();
+
+                let verify_key = VerifyingKey::from_sec1_bytes(&pubkey)
+                    .map_err(|_| MemoryError::AddressNotRelocatable)?;
+
+                let signature = signatures
+                    .get(&pubkey_addr)
+                    .ok_or(MemoryError::AddressNotRelocatable)?;
+
+                verify_key
+                    .verify(&msg, signature)
+                    .map_err(|_| MemoryError::AddressNotRelocatable)?;
+                Ok(Vec::new())
+            },
```

Comment:
These errors also don't make sense, we should create specific error types for these cases